packages/pwa-kit-react-sdk/src/ssr/universal/hooks/use-server-effect.tsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React, {useContext, useState, useEffect} from 'react'
+import {useUID} from 'react-uid' // TODO: Probably not required
+import useGetPropsArgs, {GetPropsArgs} from './use-get-props-args'
+
+// Globals
+export const DEFAULT_CONTEXT_KEY = '__SERVER_EFFECTS__'
+const contexts = []
+const isServer = typeof window === 'undefined'
+
+// Type Definitions
+type ServerEffectState = {
+    data: object;
+    loading: boolean;
+    error: Error;
+}
+type DidUpdateFn = (serverEffectArgs: GetPropsArgs) => Promise<any>
+type EffectsResolver = () => Promise<{[x: string]: {}}>
+type Provider = ({ children, value }: { children: any; value: any; }) => JSX.Element
```

Comment:
```suggestion
type Provider = (arg: { children: unknown[]; value: unknown; }) => JSX.Element
```
1. Don't need to destructure in a type, because we don't use the values.
2. Presumably `children` will be an array.
3. Prefer `unknown` to `any` for better type safety.